### ISSUE TITLE
Systemd service file

### ISSUE THREAD

**Issue by UweSauter** [adrienverge/openfortivpn#623]
You might want to include the following file as `/usr/lib/systemd/system/openfortivpn@.service`. It allows openfortivpn to act as system service with its configuration stored in `/etc/openfortivpn/<NAME>.conf`.
Start a particular configuration with `systemctl start openfortivpn@NAME.service`.

```
[Unit]
Description=OpenFortiVPN for %I
After=network-online.target

[Service]
Type=simple
PrivateTmp=true
ExecStart=/usr/bin/openfortivpn -c /etc/openfortivpn/%I.conf
OOMScoreAdjust=-100

[Install]
WantedBy=multi-user.target
```

**Comment by DimitriPapadopoulos** [comment]
Not the _openfortivpn_ sources are the place to add this file to.

If you want this file added to RPM packages please ask the packagers. I think that's where you may add a pull request:
https://src.fedoraproject.org/rpms/openfortivpn

**Comment by UweSauter** [comment]
As you say.
At least others now have a reference if they build openfortivpn manually.

**Comment by DimitriPapadopoulos** [comment]
If you think it's useful outside of vendor packages we could add the file. It's just that not all target platforms have _systemd_.

We need to detect if _systemd_ is available and find the location to write the file to. I don't know is the target always `/usr/lib/systemd/system/? Could _systemd_ be installed elsewhere, even in theory? We probably need to query `pkg-config`.

**Comment by DimitriPapadopoulos** [comment]
On Ubuntu 18.04 **`/usr/lib/systemd/system`** is exposed in by variable `systemdsystemunitpath` of **`pkg-config`** together with other paths:
```
$ pkg-config systemd --variable=systemdsystemunitpath
/etc/systemd/system:/etc/systemd/system:/run/systemd/system:/usr/local/lib/systemd/system:/lib/systemd/system:/usr/lib/systemd/system:/lib/systemd/system
$ 
```
However on _Ubuntu_ **`/usr/lib/systemd/system`** does not exist, only **`/lib/systemd/system`**:
```
$ ls /usr/lib/systemd/system
ls: cannot access '/usr/lib/systemd/system': No such file or directory
$ 
$ ls /lib/systemd/system/*@.service
/lib/systemd/system/apport-forward@.service
/lib/systemd/system/autovt@.service
/lib/systemd/system/clean-mount-point@.service
/lib/systemd/system/configure-printer@.service
[...]
/lib/systemd/system/wpa_supplicant-wired@.service
/lib/systemd/system/wpa_supplicant@.service
$ 
```
That said on _Fedora_ `/lib` is a symlink to `/usr/lib`:
```
$ ls -l /lib
lrwxrwxrwx. 1 root root 7 Jul 25  2019 /lib -> usr/lib
$ 
```
And indeed while some of these files are brought by the `systemd` package, other are brought by other packages similar to `openfortivpn`:
```
$ dpkg -S /lib/systemd/system/*@.service
apport: /lib/systemd/system/apport-forward@.service
systemd: /lib/systemd/system/autovt@.service
udisks2: /lib/systemd/system/clean-mount-point@.service
system-config-printer-udev: /lib/systemd/system/configure-printer@.service
[...]
wpasupplicant: /lib/systemd/system/wpa_supplicant@.service
wpasupplicant: /lib/systemd/system/wpa_supplicant-wired@.service
$ 
```
In short I'm OK with adding this file but I don't know how to find the target directory, either `/lib/systemd/system` or `/usr/lib/systemd/system`. I'll have a look at how other _Ubuntu_ and _Fedora_ packages generate similar files. Are they installed by upstream sources or by vendor-specific packages?

**Comment by mrbaseman** [comment]
perhaps as a template in `$PREFIX/share/systemd`?

**Comment by DimitriPapadopoulos** [comment]
The _apport_ setup does seem to be installing this file:
https://github.com/rickysarraf/apport/blob/0582b09/setup.py#L115-L122
```
# determine systemd unit directory
try:
    systemd_unit_dir = subprocess.check_output(
        ['pkg-config', '--variable=systemdsystemunitdir', 'systemd'],
        universal_newlines=True).strip()
except subprocess.CalledProcessError:
    # hardcoded fallback path
    systemd_unit_dir = '/lib/systemd/system'
```
It appears the proper _pkg-config_ variable is **`systemdsystemunitdir`**.

On _Ubuntu_ 18.04:
```
$ pkg-config --variable=systemdsystemunitdir systemd
/lib/systemd/system
$ 
```
On _Fedora_ 31:
```
$ pkg-config --variable=systemdsystemunitdir systemd
/usr/lib/systemd/system
$ 
```

We can get the source package to install the file if _systemd_ is available. But then vendor packages might need to be modified as well to include this file.

**Comment by DimitriPapadopoulos** [comment]
More information on how to use _Autotools_ to install the file here:
https://stackoverflow.com/questions/58017453/install-systemd-service-using-autotools

**Comment by UweSauter** [comment]
When I installed it a few days ago from EPEL 8 no systemd unit file was included. This is the content of the package:

```
# rpm -ql openfortivpn
/etc/openfortivpn
/etc/openfortivpn/config
/usr/bin/openfortivpn
/usr/lib/.build-id
/usr/lib/.build-id/cf
/usr/lib/.build-id/cf/0a00798e7f1acf977f7aeb9e63f9046df0980d
/usr/share/doc/openfortivpn
/usr/share/doc/openfortivpn/README.md
/usr/share/licenses/openfortivpn
/usr/share/licenses/openfortivpn/LICENSE
/usr/share/man/man1/openfortivpn.1.gz
/usr/share/openfortivpn
/usr/share/openfortivpn/config.template
```

**Comment by DimitriPapadopoulos** [comment]
@UweSauter Yes, right now it is totally expected that not systemd unit file is included.

**Comment by UweSauter** [comment]
As I wrote before I'm happy if this issue is just kept as reference for those in need of a systemd unit file.

**Comment by DimitriPapadopoulos** [comment]
@UweSauter We are precisely in the process of finding a way to add the file.

**Comment by DimitriPapadopoulos** [comment]
I have provided a patch that installs the file when _systemd_ is available:
https://github.com/adrienverge/openfortivpn/pull/633
Do you want to give it a try?
